ui/app/helpers/utils/util.js

```diff
@@ -465,18 +465,31 @@ export function constructTxParams({
  * or throws an error in case of failure.
  */
 export async function jsonRpcRequest(rpcUrl, rpcMethod, rpcParams = []) {
+  let fetchUrl = rpcUrl
+  const headers = {
+    'Content-Type': 'application/json',
+  }
+  // Convert basic auth URL component to Authorization header
+  const { origin, pathname, username, password } = new URL(rpcUrl)
+  // URLs containing username and password needs special processing
+  if (username && password) {
+    // eslint-disable-next-line no-unused-vars
+    const encodedAuth = Buffer.from(`${username}:${password}`).toString(
+      'base64',
+    )
+    headers.Authorization = `Basic ${encodedAuth}`
+    fetchUrl = `${origin}${pathname}`
```

Comment:
This is _nearly_ complete, but we should also include `search` here so that search parameters are not omitted.

```suggestion
    fetchUrl = `${origin}${pathname}${search}`
```

`hash` is missing as well, but that seems OK to omit since it's not sent over the network anyway.

Reply:
didn't think RPC urls would typically have query strings, but there's no harm in including it. Thanks for the suggestion.